ui/pages/shield-plan/shield-payment-modal.tsx

```diff
@@ -31,51 +31,52 @@ import {
 import { AssetPickerModal } from '../../components/multichain/asset-picker-amount/asset-picker-modal';
 import { TabName } from '../../components/multichain/asset-picker-amount/asset-picker-modal/asset-picker-modal-tabs';
 import { useI18nContext } from '../../hooks/useI18nContext';
-import {
-  AssetWithDisplayData,
-  ERC20Asset,
-  NativeAsset,
-} from '../../components/multichain/asset-picker-amount/asset-picker-modal/types';
-import { useMultichainBalances } from '../../hooks/useMultichainBalances';
-import { AssetType } from '../../../shared/constants/transaction';
 import {
   CHAIN_ID_TO_NETWORK_IMAGE_URL_MAP,
   CHAIN_IDS,
   NETWORK_TO_NAME_MAP,
 } from '../../../shared/constants/network';
-import {
-  PAYMENT_METHODS,
-  PaymentMethod,
-  SUPPORTED_STABLE_TOKENS,
-} from './types';
+import { TokenWithApprovalAmount } from '../../hooks/subscription/useSubscriptionPricing';
+import { PAYMENT_TYPES, PaymentType } from '@metamask/subscription-controller';
+import { AssetWithDisplayData, ERC20Asset, NativeAsset } from '../../components/multichain/asset-picker-amount/asset-picker-modal/types';
 
 export const ShieldPaymentModal = ({
   isOpen,
   onClose,
   selectedPaymentMethod,
   setSelectedPaymentMethod,
+  availableTokenBalances,
   selectedToken,
   onAssetChange,
   hasStableTokenWithBalance,
 }: {
   isOpen: boolean;
   onClose: () => void;
-  selectedPaymentMethod: PaymentMethod;
-  setSelectedPaymentMethod: (method: PaymentMethod) => void;
+  selectedPaymentMethod: PaymentType;
+  setSelectedPaymentMethod: (method: PaymentType) => void;
+  availableTokenBalances: TokenWithApprovalAmount[];
   selectedToken?:
-    | AssetWithDisplayData<ERC20Asset>
-    | AssetWithDisplayData<NativeAsset>;
+    TokenWithApprovalAmount;
   onAssetChange: (
-    asset: AssetWithDisplayData<ERC20Asset> | AssetWithDisplayData<NativeAsset>,
+    asset: TokenWithApprovalAmount,
   ) => void;
   hasStableTokenWithBalance: boolean;
 }) => {
   const t = useI18nContext();
   const [showAssetPickerModal, setShowAssetPickerModal] = useState(false);
 
-  // Get multichain balances to filter tokens with balance
-  const { assetsWithBalance: multichainTokensWithBalance } =
-    useMultichainBalances();
+  const selectPaymentMethod = useCallback(
+    (selectedMethod: PaymentType) => {
+      setSelectedPaymentMethod(selectedMethod);
```

Comment:
Mixing `PaymentMethod` with `PaymentType`. Why not use `PaymentMethod` consistently?